### PR TITLE
[HttpClient] Explain how to mock `TransportExceptions` that occur before headers are received

### PR DESCRIPTION
**Pull request by mpdude**
The `error` option value was previously undocumented.

See https://github.com/symfony/symfony/pull/57081#issuecomment-2189260078 for the discussion leading to this.

PR based on 6.4 since the section was not yet present in 5.4 docs.

### CODE REVIEW

File: http_client.rst [hunk 1]
```diff
@@ -2282,7 +2282,26 @@ when making HTTP requests you might face errors at transport level.
...
 error. :class:`Symfony\\Component\\HttpClient\\Response\\MockResponse` allows
-you to do so, by yielding the exception from its body::
+you to do so in multiple ways.
+
+In order to test errors that occur before headers have been received,
+set the ``error`` option value when creating the ``MockResponse``.
+Transport errors of this kind occur, for example, when a host name
+cannot be resolved or the host was unreachable. The
+``TransportException`` will be thrown as soon as a method like
+``getStatusCode()`` or ``getContent()`` is called.
```
Comment (nicolas-grekas) [inline http_client.rst]:
```suggestion
``getStatusCode()`` or ``getHeaders()`` is called.
```

File: http_client.rst [hunk 2]
```diff
@@ -2282,7 +2282,26 @@ when making HTTP requests you might face errors at transport level.
...
+
+In order to test errors that occur while a response is being streamed
+(that is, after the headers have already been received), provide the
+response as one of the iterable ``body`` parts from
```
Comment (nicolas-grekas) [inline http_client.rst]: > provide the response as one of the iterable

I don't understand this sentence, it may need to be reformulated :)

Comment (mpdude) [reply http_client.rst]: Did so, please re-read

**Review by nicolas-grekas — APPROVED** [review]

**Comment by javiereguiluz** [comment]
Thank you Matthias.

**Pull request merged**